### PR TITLE
update to enum DS Text component

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Add enum version of const in `design-system` file to better support newer TS component. This update focuses on changes for the `Text` component. Keeping the original const available to reduce breaking changes 

* Fixes #18581 

## Screenshots/Screencaps

No visual breaking changes 

### Before
Screenshot 2023-04-13 at 1 43 35 PM

### After
Screenshot 2023-04-13 at 1 43 39 PM

## Manual Testing Steps
`yarn jest ui/components/component-library`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/text/README.mdx [hunk 1]
```diff
@@ -538,14 +538,14 @@ import { TextVariant } from '../../../helpers/constants/design-system';
 
 The prop name `align` has been deprecated in favor of `textAlign`
 
-Values and using the `TEXT_ALIGN` object from `./ui/helpers/constants/design-system.js` remain the same
+Values and using the `TextAlign` object from `./ui/helpers/constants/design-system.js` remain the same
```
Comment (NidhiKJha) [inline ui/components/component-library/text/README.mdx:541]: Question: Shouldn't this be ?
```suggestion
Values using the `TextAlign` object from `./ui/helpers/constants/design-system.js` remain the same
```

File: ui/components/component-library/text/README.mdx [hunk 2]
```diff
@@ -538,14 +538,14 @@ import { TextVariant } from '../../../helpers/constants/design-system';
...
 
 ```jsx
 // Before
-<Typograpghy align={TEXT_ALIGN.CENTER}>Demo</Typograpghy>;
+<Typograpghy align={TextAlign.Center}>Demo</Typograpghy>;
```
Comment (NidhiKJha) [inline ui/components/component-library/text/README.mdx:545]:
```suggestion
<Typography align={TextAlign.Center}>Demo</Typography>;
```

**Review by NidhiKJha — COMMENTED** [review]
Looking really good. Just some small nits

File: ui/components/component-library/text/README.mdx [hunk 3]
```diff
@@ -112,117 +112,117 @@ import { TextColor, BackgroundColor } from '../../../helpers/constants/design-sy
...
-<Text fontStyle={FONT_STYLE.ITALIC}>
+<Text fontStyle={FontStyle.Italic}>
   bold
 </Text>
 ```
 
 ### Text Transform
 
-Use the `textTransform` prop and the `TEXT_TRANSFORM` object from `./ui/helpers/constants/design-system.js` to change the text alignment of the `Text` component
+Use the `textTransform` prop and the `TextTransform` enum from `./ui/helpers/constants/design-system.js` to change the text alignment of the `Text` component
```
Comment (DDDDDanica) [inline ui/components/component-library/text/README.mdx:165]:
```suggestion
Use the `textTransform` prop and the `TextTransform` enum from `./ui/helpers/constants/design-system.ts` to change the text alignment of the `Text` component
```

File: ui/components/component-library/text/README.mdx [hunk 4]
```diff
@@ -112,117 +112,117 @@ import { TextColor, BackgroundColor } from '../../../helpers/constants/design-sy
...
-<Text textAlign={TEXT_TRANSFORM.CAPITALIZE}>
+<Text textAlign={TextTransform.Capitalize}>
   capitalize
 </Text>
 ```
 
 ### Text Align
 
-Use the `textAlign` prop and the `TEXT_ALIGN` object from `./ui/helpers/constants/design-system.js` to change the text alignment of the `Text` component
+Use the `textAlign` prop and the `TextAlign` enum from `./ui/helpers/constants/design-system.js` to change the text alignment of the `Text` component
```
Comment (DDDDDanica) [inline ui/components/component-library/text/README.mdx:188]:
```suggestion
Use the `textAlign` prop and the `TextAlign` enum from `./ui/helpers/constants/design-system.ts` to change the text alignment of the `Text` component
```

File: ui/components/component-library/text/README.mdx [hunk 5]
```diff
@@ -112,117 +112,117 @@ import { TextColor, BackgroundColor } from '../../../helpers/constants/design-sy
...
-<Text textAlign={TEXT_ALIGN.END}>
+<Text textAlign={TextAlign.End}>
   end
 </Text>
 ```
 
 ### Overflow Wrap
 
-Use the `overflowWrap` prop and the `OVERFLOW_WRAP` object from `./ui/helpers/constants/design-system.js` to change the overflow wrap of the `Text` component
+Use the `overflowWrap` prop and the `OverflowWrap` enum from `./ui/helpers/constants/design-system.js` to change the overflow wrap of the `Text` component
```
Comment (DDDDDanica) [inline ui/components/component-library/text/README.mdx:217]:
```suggestion
Use the `overflowWrap` prop and the `OverflowWrap` enum from `./ui/helpers/constants/design-system.ts` to change the overflow wrap of the `Text` component
```

File: ui/components/component-library/text/README.mdx (hunk 1 above)
Comment (DDDDDanica) [reply ui/components/component-library/text/README.mdx:541]:
```suggestion
Values and using the `TextAlign` object from `./ui/helpers/constants/design-system.ts` remain the same
```

**Comment by DDDDDanica** [comment]
Looks good, just some small changes ! :)

**Review by brad-decker — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**